src/lucky/assignable.cr

```diff
@@ -89,7 +89,9 @@ module Lucky::Assignable
       {% sorted_assigns = ASSIGNS.sort_by { |dec|
            has_explicit_value =
              dec.type.is_a?(Metaclass) ||
-               dec.type.types.map(&.id).includes?(Nil.id) ||
+               dec.type.types.any? { |t|
+                 (t.is_a?(Metaclass) || t.is_a?(ProcNotation) || t.is_a?(Generic)) ? false : t.names.includes?(Nil.id)
```

Comment:
Not a fan of doing this, but if we use `type.resolve` here we will have to force all Lucky apps to start adding a bunch of `require` statements at the top of all the files because of examples like..

```crystal
class PostPage < MainLayout
   # in macro land, Crystal doesn't know what "User" is yet. If we resolve this type at compile time, we will have to require it at the top of this file.
  needs user : User

  # ...
end
```

This starts to compound when you're talking about hundreds of actions, models, queries, operations, pages, etc...